ncat/ncat_listen.c

```diff
@@ -621,6 +627,65 @@ int read_stdin(void)
     return nbytes;
 }
 
+/* Read from stdin and broadcast to all client sockets. Return the number of
+   bytes read, or -1 on error. */
+int read_stdin_delimiter(void)
+{
+    char buf[DEFAULT_TCP_BUF_LEN];
+
+    /* Converting the ascii valued delimiter parameter to character. */
+    char delimiter = (int)o.delimiter;
+
+    char tc[1];       /* Temporary character */
+    int index = 0;    /* Index of the array */
+    int nbytes;       /* Number of bytes to be broadcasted */
+
+    char *tempbuf = NULL;
+
+    /* Loop executes until it finds the delimiter or EOF
+      or sizeof(buf) < DEFAULT_TCP_BUF_LEN which is the maximum size. */
+    do {
+        nbytes = read(STDIN_FILENO, tc, sizeof(tc));
+        buf[index++] = tc[0];
+    } while(index < DEFAULT_TCP_BUF_LEN && tc[0] != delimiter && tc[0] != '\0' && nbytes != 0);
```

Comment:
Why the `tc[0] != '\0'` check? This prevents sending data containing null bytes, which does occur.
Also, need to check for `nbytes < 0`, which indicates an error.